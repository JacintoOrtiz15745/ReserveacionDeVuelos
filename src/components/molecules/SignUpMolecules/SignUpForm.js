import React from 'react';
import {View} from 'react-native';
import {styles} from '../../../styles/SignUpStyles';
import {SignUpConstants} from '../../../utils/Constants';
  import SignUpFormInput from '../../atoms/SignUpAtoms/SignUpFormInput';
import SignUpFormCheckBox from '../../atoms/SignUpAtoms/SignUpFormCheckBox';
import {
  handleInputNameChange,
  handleInputEmailChange,
  handleInputPasswordChange,
  handleTermsCheckboxMark,
  handleUpdatesCheckboxMark,
} from '../../../utils/Functions';
const SignUpForm = (props) => {
  const {formValues, setFormValues} = props;
  return (
    <View testID='signUpForm'>
      <SignUpFormInput
        inputName={SignUpConstants.NameInputTitle}
        value={formValues.name}
        validation={formValues.validName}
        errorMessage={SignUpConstants.nameErrorMessage}
        setFormValues={setFormValues}
        handleChange={handleInputNameChange}
        formValues={formValues}
        isValidated={formValues.validatedName}
      />
      <SignUpFormInput
        inputName={SignUpConstants.EmailInputTitle}
        value={formValues.email}
        validation={formValues.validEmail}
        errorMessage={SignUpConstants.emailErrorMessage}
        setFormValues={setFormValues}
        handleChange={handleInputEmailChange}
        formValues={formValues}
        isValidated={formValues.validatedEmail}
      />
      <SignUpFormInput
        inputName={SignUpConstants.PasswordInputTitle}
        value={formValues.password}
        validation={formValues.validPassword}
        errorMessage={SignUpConstants.passwordErrorMessage}
        setFormValues={setFormValues}
        handleChange={handleInputPasswordChange}
        formValues={formValues}
        isValidated={formValues.validatedPassword}
      />
      <View style={styles.checkboxFather}>
        <SignUpFormCheckBox
          checkBoxValue={formValues.termsCheckbox}
          formValues={formValues}
          setFormValues={setFormValues}
          checkboxMessage={SignUpConstants.TermsCheckboxTitle}
          handleChange={handleTermsCheckboxMark}
        />
        <SignUpFormCheckBox
          checkBoxValue={formValues.updatesCheckbox}
          formValues={formValues}
          setFormValues={setFormValues}
          checkboxMessage={SignUpConstants.UpdatesCheckboxTitle}
          handleChange={handleUpdatesCheckboxMark}
        />
      </View>
    </View>
  );
};

export default SignUpForm;
