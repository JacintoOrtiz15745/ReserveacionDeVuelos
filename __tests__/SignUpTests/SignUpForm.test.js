import React from 'react';
import SignUpForm from '../../src/components/molecules/SignUpMolecules/SignUpForm';
import {shallow} from 'enzyme';


const formValues = {
  name: '',
  email: '',
  password: '',
  termsCheckbox: false,
  updatesCheckbox: false,
  validName: true,
  validEmail: true,
  validPassword: true,
  validatedName: false,
  validatedEmail: false,
  validatedPassword: false,
};
const props = {
  formValues,
  setFormValues: jest.fn(),
};
let wrapper;
beforeEach(() => {
  wrapper = shallow(<SignUpForm {...props} />);
});
describe('renders correclty', () => {
  test('snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('Titles match with the num of inputs', () => {
    const component = wrapper.findWhere(
      (node) => node.prop('testID') === 'signUpForm',
    );

    expect(component.exists()).toBeTruthy();
  });
});
