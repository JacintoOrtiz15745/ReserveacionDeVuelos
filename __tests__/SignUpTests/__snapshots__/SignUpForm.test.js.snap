// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correclty snapshot 1`] = `
<View
  testID="signUpForm"
>
  <SignUpFormInput
    errorMessage=" must be at least 3 characters."
    formValues={
      Object {
        "email": "",
        "name": "",
        "password": "",
        "termsCheckbox": false,
        "updatesCheckbox": false,
        "validEmail": true,
        "validName": true,
        "validPassword": true,
        "validatedEmail": false,
        "validatedName": false,
        "validatedPassword": false,
      }
    }
    handleChange={[Function]}
    inputName="First Name"
    isValidated={false}
    setFormValues={[MockFunction]}
    validation={true}
    value=""
  />
  <SignUpFormInput
    errorMessage="invalid Email format"
    formValues={
      Object {
        "email": "",
        "name": "",
        "password": "",
        "termsCheckbox": false,
        "updatesCheckbox": false,
        "validEmail": true,
        "validName": true,
        "validPassword": true,
        "validatedEmail": false,
        "validatedName": false,
        "validatedPassword": false,
      }
    }
    handleChange={[Function]}
    inputName="Email *"
    isValidated={false}
    setFormValues={[MockFunction]}
    validation={true}
    value=""
  />
  <SignUpFormInput
    errorMessage="must be at least 8 characters"
    formValues={
      Object {
        "email": "",
        "name": "",
        "password": "",
        "termsCheckbox": false,
        "updatesCheckbox": false,
        "validEmail": true,
        "validName": true,
        "validPassword": true,
        "validatedEmail": false,
        "validatedName": false,
        "validatedPassword": false,
      }
    }
    handleChange={[Function]}
    inputName="Password *"
    isValidated={false}
    setFormValues={[MockFunction]}
    validation={true}
    value=""
  />
  <View
    style={
      Object {
        "marginBottom": 15,
      }
    }
  >
    <SignUpFormCheckBox
      checkBoxValue={false}
      checkboxMessage="I agree to the Terms and Privacy Policy."
      formValues={
        Object {
          "email": "",
          "name": "",
          "password": "",
          "termsCheckbox": false,
          "updatesCheckbox": false,
          "validEmail": true,
          "validName": true,
          "validPassword": true,
          "validatedEmail": false,
          "validatedName": false,
          "validatedPassword": false,
        }
      }
      handleChange={[Function]}
      setFormValues={[MockFunction]}
    />
    <SignUpFormCheckBox
      checkBoxValue={false}
      checkboxMessage="Subscribe for select product updates."
      formValues={
        Object {
          "email": "",
          "name": "",
          "password": "",
          "termsCheckbox": false,
          "updatesCheckbox": false,
          "validEmail": true,
          "validName": true,
          "validPassword": true,
          "validatedEmail": false,
          "validatedName": false,
          "validatedPassword": false,
        }
      }
      handleChange={[Function]}
      setFormValues={[MockFunction]}
    />
  </View>
</View>
`;
